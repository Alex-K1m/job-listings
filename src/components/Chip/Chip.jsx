import cn from 'classnames';
import { string } from 'prop-types';
import React from 'react';

import styles from './Chip.module.scss';

const Chip = ({ color, children }) => (
  <div className={cn(styles.chip, { [styles.dark]: color === 'dark' })}>
    <span className={styles.label}>{children}</span>
  </div>
);

Chip.defaultProps = {
  color: 'primary',
};

Chip.propTypes = {
  color: string,
  children: string.isRequired,
};

export default Chip;
