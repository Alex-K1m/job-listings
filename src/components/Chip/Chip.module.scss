@use '../colors' as c;

.chip {
  align-items: center;
  background-color: c.$primary;
  border-radius: 1.2rem;
  color: #fff;
  display: inline-flex;
  font: 700 1.1rem/127% Spartan, sans-serif;
  height: 2.4rem;
  letter-spacing: -0.0072em;
  padding: 0 0.8rem;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

.dark {
  background-color: c.$text;
}

.label {
  transform: translateY(0.1rem);
}
