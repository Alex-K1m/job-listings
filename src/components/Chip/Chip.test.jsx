import React from 'react';
import renderer from 'react-test-renderer';

import Chip from './Chip';

describe('Chip', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Chip>Some text</Chip>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
