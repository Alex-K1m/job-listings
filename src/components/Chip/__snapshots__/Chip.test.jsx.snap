// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip renders correctly 1`] = `
<div
  className="chip"
>
  <span
    className="label"
  >
    Some text
  </span>
</div>
`;
